sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -1583,3 +1586,145 @@ def get_trial_metrics(
+                raise RuntimeError(f"Trial {trial_name} not found or has no status")
+
+            # The job name is typically the same as trial name for batch/Job trials
+            job_name = trial_name
+
+            # Get logs from the job's pod
+            core_api = CoreV1Api()
+
+            # Find pods associated with the job
+            label_selector = f"katib.kubeflow.org/trial={trial_name}"
+            
+            # Check if trial has primaryPodLabels to identify the primary pod
+            primary_pod_labels = None
+            if (trial.get("status", {}).get("primaryPodLabels")):
+                primary_pod_labels = trial["status"]["primaryPodLabels"]
+                # Build label selector from primaryPodLabels
+                primary_selectors = [f"{k}={v}" for k, v in primary_pod_labels.items()]
+                label_selector = ",".join([label_selector] + primary_selectors)
+            
+            pods = core_api.list_namespaced_pod(
+                namespace=namespace, label_selector=label_selector
+            )
+
+            if not pods.items:
+                # Fallback: try without primaryPodLabels if no pods found
+                if primary_pod_labels:
+                    label_selector = f"katib.kubeflow.org/trial={trial_name}"
+                    pods = core_api.list_namespaced_pod(
+                        namespace=namespace, label_selector=label_selector
+                    )
+                
+                if not pods.items:
+                    raise RuntimeError(f"No pods found for trial {trial_name}")
+
+            # Get the primary pod (should be the only one if primaryPodLabels worked correctly)
+            pod = pods.items[0]
+            pod_name = pod.metadata.name
+
+            # Determine container name based on metrics collector and trial spec
+            if container is None:
+                # Get the experiment to check metrics collector type
+                experiment = self.get_experiment(name, namespace)
+                metrics_collector_kind = None
+                if (experiment.spec.metrics_collector_spec and 
+                    experiment.spec.metrics_collector_spec.collector):
+                    metrics_collector_kind = experiment.spec.metrics_collector_spec.collector.kind
+                
+                # Determine container based on metrics collector type
+                if metrics_collector_kind == "StdOut":
+                    # For StdOut metrics collector, logs are in the sidecar container
+                    container = "metrics-logger-and-collector"
+                elif metrics_collector_kind == "Push":
+                    # For Push metrics collector, logs are in the primary container
+                    if trial.get("spec", {}).get("primaryContainerName"):
+                        container = trial["spec"]["primaryContainerName"]
+                    elif pod.spec.containers:
+                        container = pod.spec.containers[0].name
```

Comment:
Primary container name will always exists in Trial spec, so you can remove this else.
```suggestion
                    container = trial["spec"]["primaryContainerName"]
```